chain/chain/src/update_shard.rs

```diff
@@ -71,19 +55,40 @@ pub enum ApplyChunkResult {
 /// State roots of split shards which are ready.
 type SplitStateRoots = HashMap<ShardUId, StateRoot>;
 
+pub(crate) struct NewChunkData {
+    pub chunk: ShardChunk,
+    pub receipts: Vec<Receipt>,
+    pub split_state_roots: Option<SplitStateRoots>,
+    pub block: ApplyTransactionsBlockContext,
+    pub is_first_block_with_chunk_of_version: bool,
+}
+
+pub(crate) struct OldChunkData {
```

Comment:
I find the name 'OldChunkData' slightly misleading. Does it mean that it's from the past block? or invalid as there is a 'new chunk data that will replace this one'?

Reply:
Yeah, I agree, this name is terrible, but that is how it was named for a long time, so I'm just sticking to it. I will rename it as part of point 2 in https://github.com/near/nearcore/issues/10261.